src/libs/Localize/index.ts

```diff
@@ -93,11 +93,46 @@ function translate<TKey extends TranslationPaths>(desiredLanguage: 'en' | 'es' |
     throw new Error(`${phraseKey} was not found in the default language`);
 }
 
+/**
+ * Map to store translated values for each locale
+ * This is used to avoid translating the same phrase multiple times.
+ *
+ * The data is stored in the following format:
+ *
+ * {
+ *  "name_en": "Name",
+ *  "name_es": "Nombre",
+ * }
+ *
+ * Note: We are not storing any translated values for phrases with variables,
+ * as they have higher chance of being unique, so we'll end up wasting space
+ * in our cache.
+ */
+const translatedValues = new Map<string, string>();
+
 /**
  * Uses the locale in this file updated by the Onyx subscriber.
  */
 function translateLocal<TKey extends TranslationPaths>(phrase: TKey, ...variables: PhraseParameters<Phrase<TKey>>) {
-    return translate(BaseLocaleListener.getPreferredLocale(), phrase, ...variables);
+    const preferredLocale = BaseLocaleListener.getPreferredLocale();
+    const key = `${phrase}_${preferredLocale}`;
+    const isVariablesEmpty = variables.length === 0;
+
+    // Directly access and assign the translated value from the cache, instead of
+    // going through map.has() and map.get() to avoid multiple lookups.
+    const valueFromCache = translatedValues.get(key);
+
+    // If the phrase is already translated, return the translated value
+    if (valueFromCache) {
+        return valueFromCache;
+    }
+    const translatedText = translate(preferredLocale, phrase, ...variables);
+
+    // We don't want to store translated values for phrases with variables
+    if (isVariablesEmpty) {
+        translatedValues.set(key, translatedText);
```

Comment:
NAB but I wonder if it would make sense to do the cache set operation async so that it doesn't delay the return of `translatedText`:

```suggestion
        setImmediate(() => translatedValues.set(key, translatedText));
```

Reply:
I think let's not introduce fixes for anything which has not performed worse as of yet. There's a chance that this might introduce any regressions as well 🤔